src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -108,7 +112,9 @@ protected void handleMessageInternal(final Message<?> message) throws Exception
 				? this.messageKeyExpression.getValue(this.evaluationContext, message)
 				: message.getHeaders().get(KafkaHeaders.MESSAGE_KEY);
 
-		this.kafkaProducerContext.send(topic, partitionId, messageKey, message.getPayload());
+		Future<RecordMetadata> future = this.kafkaProducerContext.send(topic, partitionId, messageKey,
```

Comment:
Welcome back, @mbogoevici !

+1 for all comments.
Just was going to say the same.
Although I'm not sure in the `KafkaTemplate` yet...